sky/backends/cloud_vm_ray_backend.py

```diff
@@ -1004,6 +1004,10 @@ def _retry_region_zones(self,
         # Get previous cluster status
         prev_cluster_status = backend_utils.refresh_cluster_status_handle(
             cluster_name, acquire_per_cluster_status_lock=False)[0]
+        prev_cluster_exists = prev_cluster_status in [
```

Comment:
I think this follows the logic in the previous code that any cluster in INIT status will be able to fail over to other regions.

However, after thinking more about this, here is the pros and cons for the behavior of adding INIT in the list as well:

When a cluster exists and is in INIT mode.

## sky start -c cluster
### Current behavior:
1. try to launch the cluster with the same spec.
2. If `ray up` fails, we terminate the cluster
3. Relaunch with the same resources as `cluster` (same region/zone/accelerators)

### New behavior
Adding INIT to the list (or use the cluster_exists directly)
1. try to launch the cluster with the same spec.
2. If `ray up` fails, we try to stop the cluster
3. Not failover, and print out not able to launch

## sky launch -c cluster
### Current behavior:
1. try to launch the cluster with the same spec.
2. If `ray up` fails, we terminate the cluster
3. Relaunch and failover with empty resources, i.e. CPU instance, no matter what the previous cluster had

### New behavior
Adding INIT to the list
1. try to launch the cluster with the same spec.
2. If `ray up` fails, we try to stop the cluster
3. Not failover, and print out not able to launch

## sky launch -c cluster task.yaml
### Current behavior
1. try to launch the cluster with the same spec.
2. If `ray up` fails, we terminate the cluster
3. Relaunch and failover with the task.resources

### New behavior
Adding INIT to the list
1. try to launch the cluster with the same spec.
2. If `ray up` fails, we try to stop the cluster
3. Not failover, and print out not able to launch

Pro for adding INIT: more consistent for the three commands
Con for adding INIT: any INIT cluster in the status table needs to explicitly `sky down` before the failover can work.
  For example, if a user Ctrl+C the `sky launch` during failover, she will have to do `sky down` first before `sky launch` being able to failover to other regions/clouds.

Based on the discussion above, I would prefer to add INIT to the list as well, to make the failover more conservative avoiding accident termination of the user's cluster.

Reply:
Agreed with going with the new behavior, with some asterisks to discuss in the future.

For example, it's a bit unintuitive to me that a `start/launch` may transition `INIT` to `STOPPED`. I see why that may be desired, but it may make the state transitions more complex. TBD.